### PR TITLE
Feature 1

### PR DESCRIPTION
**Pull request by misdaque**
Added Services to onboard devices and rooms...

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,4 +23,5 @@
 hs_err_pid*
 .idea
 target
-*.iml
\ No newline at end of file
+*.iml
+deploy
```
Comment (keensam04) [inline .gitignore]: remove deploy folder from being ignored

File: src/main/java/com/autohome/dao/DeviceRepo.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        return null;
+    }
+
+    public List<Device> getDevices(int roomId) {
+        String query = "SELECT * FROM device WHERE roomId = ?";
+       try {
+           List<Device> listOfDevices = jdbcTemplate.query(query, new Object[]{roomId}, new RowMapper<Device>() {
+
+               @Override
+               public Device mapRow(ResultSet rs, int rowNum) throws SQLException {
```
Comment (keensam04) [inline src/main/java/com/autohome/dao/DeviceRepo.java]: refactor redundant code

File: src/main/java/com/autohome/dao/RoomRepo.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                return null;
+            }
+        } catch (EmptyResultDataAccessException e) {
+            // this is a valid exception. indicates that roomName is not present in the table
+            log.warn("Room with Id {} not present",Id);
+        }
+        return null;
+    }
+    //Sql Injection;;
+    public boolean updateRoom(Room room, int id){ // roomName = hall;delete * from device;
```
Comment (keensam04) [inline src/main/java/com/autohome/dao/RoomRepo.java]: remove comment

File: src/main/java/com/autohome/dao/RoomRepo.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        try {
+            return jdbcTemplate.query(query, roomMapper());
+        }
+        catch (EmptyResultDataAccessException e){
+            log.warn("No rooms found");
+        }
+        return null;
+    }
+
+    private Room mapRoom(ResultSet rs, int rowNum) throws SQLException {
```
Comment (keensam04) [inline src/main/java/com/autohome/dao/RoomRepo.java]: remove method

File: src/main/java/com/autohome/service/RoomService.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        return roomFromDb != null;
+    }
+
+    public Room getRoom(int id) {
+        return roomRepo.getRoom(id);
+    }
+
+    public boolean updateRoom(Room room, int id) {
+        if (isRoomPresent(id)){
+            log.info("Room with id {} has noot been updated",room.getId());
```
Comment (keensam04) [inline src/main/java/com/autohome/service/RoomService.java]: not spelling wrong

**Review by keensam04 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (misdaque) [reply .gitignore]: Resolved

File: src/main/java/com/autohome/dao/DeviceRepo.java (hunk 2 above)
Comment (misdaque) [reply src/main/java/com/autohome/dao/DeviceRepo.java]: Resolved

File: src/main/java/com/autohome/dao/RoomRepo.java (hunk 3 above)
Comment (misdaque) [reply src/main/java/com/autohome/dao/RoomRepo.java]: Resolved

File: src/main/java/com/autohome/dao/RoomRepo.java (hunk 4 above)
Comment (misdaque) [reply src/main/java/com/autohome/dao/RoomRepo.java]: Resolved

File: src/main/java/com/autohome/service/RoomService.java (hunk 5 above)
Comment (misdaque) [reply src/main/java/com/autohome/service/RoomService.java]: Resolved

File: pom.xml [hunk 6]
```diff
@@ -29,6 +29,17 @@
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.18</version>
```
Comment (keensam04) [inline pom.xml]: move version to properties section

**Review by keensam04 — APPROVED** [review]

**Pull request merged**